pkgs/development/libraries/glibc/default.nix

```diff
@@ -119,6 +119,16 @@ callPackage ./common.nix { inherit stdenv; } {
 
       # Get rid of more unnecessary stuff.
       rm -rf $out/var $bin/bin/sln
+
+      # Backwards-compatibility to fix e.g.
+      # "configure: error: Pthreads are required to build libgomp" during `gcc`-build
+      # because it's not actually needed anymore to link against `pthreads` since
+      # it's now part of `libc.so.6` itself, but the gcc build breaks if
+      # this doesn't work.
+      ln -sf $out/lib/libpthread.so.0 $out/lib/libpthread.so
```

Comment:
Upstream recommends installing empty `.a` files like `libpthread.a` for future bianries to avoid dynamic linking against these libraries.

Reply:
Done, change will be part of the next push.

However, this doesn't fix the initrd issue on its own: `${pkgs.kmod}` doesn't have `libpthread` in its `DT_NEEDED` and thus doesn't seem to be linked against it anymore. Also, due to no other references, `libpthread.a` isn't part of `$out/lib` from `extra-utils`, so the hack is still needed IMO.